persist/sqlite/consensus_refactored_test.go

```diff
@@ -570,6 +600,78 @@ func TestV2TransactionChainIndices(t *testing.T) {
 	n.assertV2ChainIndices(t, txn2.ID())
 }
 
+func TestMaturedSiacoinBalance(t *testing.T) {
+	pk1 := types.GeneratePrivateKey()
+	uc1 := types.StandardUnlockConditions(pk1.PublicKey())
+	addr1 := uc1.UnlockHash()
+
+	n := newTestChain(t, false, nil)
+
+	checkBalance := func(addr types.Address, expectedSC, expectedImmatureSC types.Currency) {
+		t.Helper()
+
+		sc, immatureSC, sf, err := n.db.Balance(addr)
+		if err != nil {
+			t.Fatal(err)
+		}
+		testutil.Equal(t, "siacoins", expectedSC, sc)
+		testutil.Equal(t, "immature siacoins", expectedImmatureSC, immatureSC)
+		testutil.Equal(t, "siafunds", 0, sf)
+	}
+
+	checkBalance(addr1, types.ZeroCurrency, types.ZeroCurrency)
+
+	b := testutil.MineBlock(n.tipState(), nil, addr1)
+	n.applyBlock(t, b)
+
+	val := b.MinerPayouts[0].Value
+
+	for range n.network.MaturityDelay {
+		checkBalance(addr1, types.ZeroCurrency, val)
+		n.mineTransactions(t)
+	}
+
+	checkBalance(addr1, val, types.ZeroCurrency)
+
+	for range n.network.MaturityDelay {
+		n.revertBlock(t)
+		checkBalance(addr1, types.ZeroCurrency, val)
+	}
+}
+
+func TestEventPayout(t *testing.T) {
+	pk1 := types.GeneratePrivateKey()
+	uc1 := types.StandardUnlockConditions(pk1.PublicKey())
+	addr1 := uc1.UnlockHash()
+
+	n := newTestChain(t, false, nil)
+
+	b := testutil.MineBlock(n.tipState(), nil, addr1)
+	n.applyBlock(t, b)
+
+	{
+		events, err := n.db.AddressEvents(addr1, 0, math.MaxInt64)
+		if err != nil {
+			t.Fatal(err)
+		}
+		testutil.Equal(t, "events", 1, len(events))
+
+		ev0 := events[0].Data.(explorer.EventPayout)
+		testutil.Equal(t, "event 0 output ID", n.tipState().Index.ID.MinerOutputID(0), ev0.SiacoinElement.ID)
+		testutil.Equal(t, "event 0 output source", explorer.SourceMinerPayout, ev0.SiacoinElement.Source)
+	}
+
+	n.revertBlock(t)
+
+	{
```

Comment:
still don't quite get the code blocks but these tests are very readable and easy to follow now 👍

Reply:
I guess we could remove them, it's mostly so I don't have to use different variables names and to logically isolate each set of check since tend to be a couple lines long.  Ideally we'd have functions like `assertTransactionsNotExist` or `assertNoEvents` but it doesn't really seem right if we only use it once.